umbraco/Umbraco-CMS
# MediaPicker3 should include media crops and focal point in output

### Prerequisites


### Description

When rendering for the Delivery API, MediaPicker3 currently only outputs local crops and focal point. Crops and focal point on the media item itself are disregarded, which is a bug.

This PR ensures we merge the property (local) crops + focal point and the media (global) crops + focal point when outputting. The local ones takes precedence over the global ones.

In this PR I have also refactored the unit test for MediaPicker3 a bit to avoid repeating the same assertion blocks over and over again.

### Testing this PR

1. Setup the "image" media type with multiple crops.
2. Setup a media with crops and focal point defined.
3. Create a content type with a MediaPicker3 property that local crops and focal point enabled.
   - make sure you setup both unique crop aliases and a crop alias that overlap one of the media ones.
4. Create content of said content type with crops and focal point configured within the property.
5. Fetch the content using the Delivery API (i.e. fetch by ID).
6. Verify that the global crops are included in the output alongside the local ones.
7. Verify that the local overlapping crop is included in the output instead of the global one.
8.  Verify that the local focal point is included in the output.
9. Repeat steps 4 through 8 but without a local focal point defined, and verify that the global focal point is included in the output.
